Shared/Samples/Geocode offline/GeocodeOfflineView.Model.swift

```diff
@@ -0,0 +1,206 @@
+            
+            // Determine the amount by which we will need to offset the callout along y-axis.
+            return CGPoint(x: .zero, y: -image.size.height)
+        }()
+        
+        /// The locator task used to preform the geocode operations, loaded from a local file.
+        private let locatorTask = LocatorTask(name: "SanDiego_StreetAddress", bundle: .main)
+        
+        /// The viewpoint of the map view, initially centered on San Diego, CA, USA.
+        @Published var viewpoint = Viewpoint(
+            center: Point(x: -13_042_250, y: 3_857_970, spatialReference: .webMercator),
+            scale: 2e4
+        )
+        
+        /// The placement of the callout on the map.
+        @Published var calloutPlacement: CalloutPlacement?
+        
+        /// A Boolean value indicating whether the callout placement should be offsetted, e.g. when the map magnifier is showing.
+        @Published var calloutShouldOffset = false
+        
+        /// The text shown in the callout.
+        @Published private(set) var calloutText: String = ""
+        
+        /// A Boolean value indicating whether to show an error alert.
+        @Published var isShowingErrorAlert = false
+        
+        /// The error shown in the error alert.
+        @Published private(set) var error: Error? {
+            didSet { isShowingErrorAlert = error != nil }
+        }
+        
+        init() {
+            graphicsOverlay.addGraphic(markerGraphic)
+        }
+        
+        // MARK: Methods
+        
+        /// Geocodes a given address and adds a marker with the corresponding address at the result's location.
+        /// - Parameter address: The given text address to geocode.
+        func geocodeSearch(address: String) async {
+            guard let locatorTask else { return }
+            
+            // Create geocode parameters.
+            let geocodeParameters = GeocodeParameters()
+            geocodeParameters.addResultAttributeName("Match_addr")
+            geocodeParameters.minScore = 75
+            
+            do {
+                // Perform geocode using the locator task with the text address and parameters.
+                let geocodeResults = try await locatorTask.geocode(
+                    forSearchText: address,
+                    using: geocodeParameters
+                )
+                
+                if let result = geocodeResults.first,
+                   let displayLocation = result.displayLocation {
+                    // If a result is found, place a marker at the result's location.
+                    let resultText = result.attributes["Match_addr"] as? String ?? ""
+                    updateMarker(to: displayLocation, withText: resultText)
+                    
+                    // Zoom to the extent of the result's location.
+                    viewpoint = Viewpoint(boundingGeometry: displayLocation.extent)
```

Comment:
Can you refactor this method a bit so that the viewpoint is moved to the view itself?